### PR TITLE
Add search for tags, title, author

### PR DESCRIPTION
**Pull request by stephanernst**
Issue #21

### CODE REVIEW

File: src/main/java/ama/home/HomeController.java [hunk 1]
```diff
@@ -27,6 +32,9 @@
     private SubmissionPostRepository submissionPostRepository;
 
     @Autowired
+    private UserRepository userRepository;
```
Comment (RyanBillard) [inline src/main/java/ama/home/HomeController.java]: is this used anywhere?

File: src/main/java/ama/home/HomeController.java [hunk 2]
```diff
@@ -37,9 +45,30 @@ public String home(Model model) {
...
     private List<SubmissionPost> getAllSubmissions() {
         List<SubmissionPost> posts = submissionPostRepository.findAll();
         posts.sort(new SubmissionPostComparator());
         return posts;
     }
+
+    private List<SubmissionPost> getFoundSubmissions(String search) {
+        List<SubmissionPost> foundPosts = new ArrayList<>();
+        if(!search.equals("")) {
+            HashSet<String> searchWords = new HashSet<>(Arrays.asList(search.split(",")));
```
Comment (RyanBillard) [inline src/main/java/ama/home/HomeController.java]: should this split by a space?

File: src/main/java/ama/post/PostController.java [hunk 3]
```diff
@@ -47,9 +44,16 @@ public String postAMASubmission(@ModelAttribute("submissionPost") SubmissionPost
             return "createsubmission";
         } else {
             post.setOp(user);
+            HashSet<String> tags = new HashSet<>();
             if(!postTags.equals("")) {
-                post.setTags(new HashSet<>(Arrays.asList(postTags.split(","))));
+                tags = new HashSet<>(Arrays.asList(postTags.split(",")));
```
Comment (AboudS777) [inline src/main/java/ama/post/PostController.java]: As mentioned by Ryan, split should be a space, not a comma

File: src/main/java/ama/home/HomeController.java [hunk 4]
```diff
@@ -37,9 +45,30 @@ public String home(Model model) {
...
+    }
+
     private List<SubmissionPost> getAllSubmissions() {
         List<SubmissionPost> posts = submissionPostRepository.findAll();
         posts.sort(new SubmissionPostComparator());
         return posts;
     }
+
+    private List<SubmissionPost> getFoundSubmissions(String search) {
+        List<SubmissionPost> foundPosts = new ArrayList<>();
```
Comment (RyanBillard) [inline src/main/java/ama/home/HomeController.java]: Should this be a set so we don't return the same post multiple times?

File: src/main/java/ama/home/HomeController.java (hunk 1 above)
Comment (SarrankanPathmanathan) [inline src/main/java/ama/home/HomeController.java]: nope.

**Pull request merged**